lib/requestResponseLink.ts

```diff
@@ -77,7 +77,7 @@ export class RequestResponseLink implements ReqResLink {
    */
   sendRequest(request: AmqpMessage, options?: SendRequestOptions): Promise<AmqpMessage> {
     if (!request) {
-      throw new Error("request is a required parameter and must be of type 'object'.");
+      throw new TypeError("request is a required parameter.");
```

Comment:
We can skip this check. End user will not be using this directly. The intermediate user will get error a few lines below when properties of the `request` object is accessed

Reply:
I agree, but even for internal usage, I think we should keep this check. As you said, the check a few lines below is for specific properties of the `request` object and I prefer to have an explicit check for the `request` object itself.

Reply:
When the below code is run and `request` is null or undefined, error will be thrown. What is the advantage of checking `request` explicitly?

```
if (count !== 1) { 
        request.message_id = generate_uuid();
} else if (!request.message_id) {
        request.message_id = generate_uuid();		
}
```

Reply:
As discussed offline - I understand that but it goes back to our discussion that I believe we should have an explicit validation for our input. 

 As in this case, what if the condition `count !== 1` is false, then it would check `request.mesage_id` which would led to a similar `<something> does not exist on undefined` error.